.github/workflows/generate_collection.yml

```diff
@@ -0,0 +1,23 @@
+name: Generate Postman Collection
+
+on:
+  pull_request:
+    branches: [ main ]
+    types: [opened, synchronize, reopened, edited]
+
+jobs:
+  generate_collection:
+    runs-on: ubuntu-latest
+    name: Generate Collection
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@v2
+
+      - name: Run collection generation script
+        run: ./postman/generate_collection
+
+      - name: Archive collection artifact
+        uses: actions/upload-artifact@v2
```

Comment:
The intention with archiving here was to make the collection artifact available to append to releases. I'm interested what you think: I was archiving collection artifacts here so there would be a series of pre-release collection artifacts available, but archiving a collection for every push to a PR does seem excessive. Does it sound better to archive a collection after a PR is merged into `main`, or just as part of a release workflow? @telday @izgeri

Reply:
I was thinking that we would have one manual run action that would generate a tarball of the spec files to be appended to a release, I would think that this could also be run in that same action. I don't see why we would need to store unnecessary artifacts, I haven't really worked with releases before though so I would be interested to hear what @izgeri / @BradleyBoutcher think.

Reply:
@izgeri and I talked about it yesterday in another thread, should've posted here. On PRs, generate but do not archive, on release, generate and archive.